File: 0041-first-missing-positive/0041-first-missing-positive.py
class Solution(object):
    def firstMissingPositive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        numbers = set()
        
        for i in range(len(nums)+1):
            numbers.add(i+1)
            
        for num in nums:
            if num in numbers:
                numbers.remove(num)
                
        return list(numbers)[0]

File: 0041-first-missing-positive/0041-first-missing-positive.cs
public class Solution {
    public int FirstMissingPositive(int[] nums) {
        
        HashSet<int> numbers = new HashSet<int>();
        
        for (int i = 0; i <= nums.Length; i++){
            numbers.Add(i+1);
        }
        
        for (int i = 0; i <nums.Length; i++){
            if (numbers.Contains(nums[i])){
                numbers.Remove(nums[i]);
            }
        }
        
        return numbers.First();
        
        
    }
}